### PR TITLE
Feature/rm

### PR DESCRIPTION
**Pull request by apmccartney**
When you see the `reichMoore/Type/test/Type.test` performance:
great scott

### CODE REVIEW

**Comment by apmccartney** [comment]
also, you probably want to build on a desktop...

File: src/resonanceReconstruction/breitWigner/singleLevel/Type/src/ctor2.hpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Type( std::vector< lvalue::Type >&& lvalues,
```
Comment (jlconlin) [inline src/resonanceReconstruction/breitWigner/singleLevel/Type/src/ctor2.hpp]: Why a separate file?

File: src/resonanceReconstruction/reichMoore/Base/src/neutronWaveNumber.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Quantity< InvRootBarns >
+neutronWaveNumber( const Quantity< ElectronVolts > energy ) const {
+  using Units4Constant =
+    decltype( pow( Barns() * ElectronVolts(), Ratio<-1,2> ) );
+
+  constexpr Quantity< Units4Constant >
```
Comment (jlconlin) [inline src/resonanceReconstruction/reichMoore/Base/src/neutronWaveNumber.hpp]: What is `Units4Constant`? I've seen it in a number of places.

**Comment by jlconlin** [comment]
Why should this be built without threading?

**Comment by apmccartney** [comment]
> Why should this be built without threading?

This library is memory intensive to build. If you run `make -j12` you'll likely run out of memory while compiling. GCC fails gracefully giving a helpful error message while Clang crashes.

File: src/resonanceReconstruction/reichMoore/Base/src/neutronWaveNumber.hpp (hunk 2 above)
Comment (apmccartney) [reply src/resonanceReconstruction/reichMoore/Base/src/neutronWaveNumber.hpp]: the definition is three lines above

File: src/resonanceReconstruction/breitWigner/multilevel/Type/src/evaluate2.hpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+template< typename... Args >
+auto evaluate( const Quantity<ElectronVolts> energy,
```
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction/breitWigner/multilevel/Type/src/evaluate2.hpp]: This file is included into the body of a class template. There are two such class template specializations, each with a distinct definition of its ctor and evaluate methods.

File: src/resonanceReconstruction/breitWigner/Type.hpp [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Type {
```
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction/breitWigner/Type.hpp]: `breitWigner::Type` is the common state between `breitWigner::singleLevel::Type` and `breitWigner::multilevel::Type`

File: src/resonanceReconstruction/breitWigner/Type/src/neutronWaveNumber.hpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
+Quantity< InvRootBarns >
+neutronWaveNumber( const Quantity< ElectronVolts > energy ) const {
+  using Units4Constant =
+    decltype( pow( Barns() * ElectronVolts(), Ratio<-1,2> ) );
+
+  constexpr Quantity< Units4Constant >
+    neutronConstant = 5.787793139E-14 * root( kilo(grams) ) / constant::dirac;
```
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction/breitWigner/Type/src/neutronWaveNumber.hpp]: sqrt( mass of neutron ) / h_bar.

Comment incoming

File: src/resonanceReconstruction.hpp [hunk 6]
```diff
@@ -1,29 +1,52 @@
...
 #include "dimwits.hpp"
 
+#include <Eigen/Dense>
+
 namespace njoy {
 namespace resonanceReconstruction {
 
 using namespace dimwits;
 
+constexpr double pi = 3.141592653589793;
```
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction.hpp]: We should make a repository!

File: src/resonanceReconstruction/breitWigner/CrossSection.hpp [hunk 7]
```diff
@@ -0,0 +1,29 @@
+struct CrossSection {
+  Quantity< Barn > elastic;
+  Quantity< Barn > capture;
+  Quantity< Barn > fission;
+
+  template< typename T1, typename T2, typename T3 >
+  CrossSection( const Pack< T1, T2, T3 >& pack ) : elastic( std::get<0>(pack) ),
+                                                   capture( std::get<1>(pack) ),
+                                                   fission( std::get<2>(pack) ){}
```
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction/breitWigner/CrossSection.hpp]: ;-( indeed. Fix incoming

File: src/resonanceReconstruction/breitWigner/Type.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction/breitWigner/Type.hpp]: Suggestions?

File: src/resonanceReconstruction/breitWigner/multilevel/src/D.hpp [hunk 8]
```diff
@@ -0,0 +1,13 @@
+inline double D( const double I, const int l ){
+  static constexpr const std::array< double, 40 > cache =
+  {{ 0., 0., 0., 0., 0.,
+     0., 0.75, 1.25, 1.75, 2.25 ,
+     0., 1., 1.666666666666667, 2.333333333333333, 3.,
```
[the remark this answers is not in the document]
Comment (apmccartney) [reply src/resonanceReconstruction/breitWigner/multilevel/src/D.hpp]: I count 16 sig figs. A double had 15.95 sig figs, so that seems alright

**Pull request merged**